wallet-new/test/unit/Test/Spec/Translation.hs

```diff
@@ -147,21 +144,31 @@ example1 GenesisValues{..} = OldestFirst [OldestFirst [t3, t4]]
              , trExtra = ["t3"]
              }
 
-    fee4 = estimateCardanoFee 1 1
-    t4   = Transaction {
+    t4   = feeFixpt txFee $ \fee4 -> Transaction {
                trFresh = 0
              , trFee   = fee4
              , trHash  = 4
              , trIns   = Set.fromList [ Input (hash t3) 1 ]
-             , trOuts  = [ Output r2 (initR0 - 1000 - fee3 - fee4) ]
+             , trOuts  = [ Output r2 (initR0 - 1000 - trFee t3 - fee4) ]
              , trExtra = ["t4"]
              }
 
+-- | Use this to construct a transaction where parts of the transaction depend on
+--   the transaction's final fee.
+feeFixpt :: (Int -> [Value] -> Value) -> (Value -> Transaction h a) -> Transaction h a
+
+feeFixpt getFee make = fst $
+    until (\(tx, idx) -> trFee tx == computedFee tx || idx >= 100) step (make 0, 0)
+  where
+    computedFee tx = getFee (length $ trIns tx) (map outVal $ trOuts tx)
+    step (tx, idx) = (make $ computedFee tx, idx + 1 :: Int)
+
 {-------------------------------------------------------------------------------
   Verify chain
 -------------------------------------------------------------------------------}
 
-intAndVerifyPure :: (GenesisValues GivenHash -> Chain GivenHash Addr)
+intAndVerifyPure :: HasGenesisBlockVersionData
+                 => (GenesisValues GivenHash -> Chain GivenHash Addr)
```

Comment:
Given that you added the fee policy to `GenesisValues`, is this `HasGenesisBlockVersionData` will required? If so, why?

Reply:
It's due to the use of `genesisValues`, which constructs the `GenesisValues` value. It does so by setting the `txFee` field to `estimateCardanoFee`, and `estimateCardanoFee` uses the `GenesisBlockVersionData` part of the configuration to get the correct parameters for the linear fee computation.